File: app/src/main/java/com/apanda/base/Module/Gank/contract/GankContract.java
package com.apanda.base.Module.Gank.contract;


import com.apanda.base.Entity.GankBean;
import com.apanda.base.Module.Gank.bean.DayBean;
import com.apanda.base.Module.Gank.bean.HisBean;
import com.apanda.base.base.BaseModel;
import com.apanda.base.base.BasePresenter;
import com.apanda.base.base.BaseView;

import rx.Observable;

/**
 * Created by apanda on 2016/10/21.
 */

public class GankContract {
    public interface Model extends BaseModel {
        Observable<GankBean> lodeGankBean();

        Observable<HisBean> loadHistory();

        Observable<DayBean> loadDayBean(String _history);
    }

    public interface View extends BaseView {
        void returnGankBean(GankBean _gankBean);

        void returnHistory(String History);

        void returnDayBean(DayBean _dayBean);
    }

    public abstract static class Presenter extends BasePresenter<View, Model> {
        public abstract void lodeGankBeanRequest();

        public abstract void loadDayBean(String _history);
    }

}

File: app/src/main/java/com/apanda/base/Module/Gank/contract/GankContract.kt
package com.apanda.base.Module.Gank.contract


import com.apanda.base.Entity.GankBean
import com.apanda.base.Module.Gank.bean.DayBean
import com.apanda.base.Module.Gank.bean.HisBean
import com.apanda.base.base.BaseModel
import com.apanda.base.base.BasePresenter
import com.apanda.base.base.BaseView

import rx.Observable

/**
 * Created by apanda on 2016/10/21.
 */

class GankContract {
    interface Model : BaseModel {
        fun lodeGankBean(): Observable<GankBean>

        fun loadHistory(): Observable<HisBean>

        fun loadDayBean(_history: String): Observable<DayBean>
    }

    interface View : BaseView {
        fun returnGankBean(_gankBean: GankBean)

        fun returnHistory(History: String)

        fun returnDayBean(_dayBean: DayBean)
    }

    abstract class Presenter : BasePresenter<View, Model>() {
        abstract fun lodeGankBeanRequest()

        abstract fun loadDayBean(_history: String)
    }

}
